core/src/main/scala/kafka/server/RemoteLeaderEndPoint.scala

```diff
@@ -203,7 +203,7 @@ class RemoteLeaderEndPoint(logPrefix: String,
       None
     } else {
       val metadataVersion = metadataVersionSupplier()
-      val version: Short = if (metadataVersion.fetchRequestVersion >= 13 && !fetchData.canUseTopicIds) {
+      val version: Short = if (!fetchData.canUseTopicIds) {
```

Comment:
This is interesting. IIRC, the topic id of fetch is introduced in 3.1, so `canUseTopicIds` should be always enabled after we bump the baseline of MV to 3.3.

Reply:
You are right. file https://issues.apache.org/jira/browse/KAFKA-19065

Reply:
As part of that ticket, we should also remove the `getOrElse` below and change `PartitionFetchState.topicId` not to be Optional, etc.

> fetchState.topicId.getOrElse(Uuid.ZERO_UUID),